### ISSUE TITLE
Question regarding learning rate scheduler

### ISSUE THREAD

**Issue by abeyang00** [facebookresearch/dino#124]
Thank you for the wonderful work.

I just have one silly question regarding learning rate scheduler

 
 lr_schedule = utils.cosine_scheduler(
        args.lr * (args.batch_size_per_gpu * utils.get_world_size()) / 256.,  # linear scaling rule
        args.min_lr,
        args.epochs, len(data_loader),
        warmup_epochs=args.warmup_epochs,
    )

What is the reason behind dividing by 256?
I saw on the other code that the author divides by 512...
I feel like that number isn't the batch size

Thanks in advance

**Comment by woctezuma** [comment]
This code appears at two locations:

https://github.com/facebookresearch/dino/blob/ad3fe6258d613aa27690e5a438541b6eaebf6d8f/main_dino.py#L237-L243

https://github.com/facebookresearch/dino/blob/9085367cf5186b29f69c85eb7fdacc3057094031/eval_linear.py#L95-L101

It is a linear scaling of the learning rate based on the batch size.

`utils.get_world_size()` returns the ["number of processes participating in the job"](https://pytorch.org/docs/stable/distributed.html). So `(args.batch_size_per_gpu * utils.get_world_size())` should be equivalent to the batch size.

If the batch size is 256, then the learning rate will be equal to `args.lr`. Otherwise, it will be linearly scaled.

**Comment by abeyang00** [comment]
@woctezuma thank you!

**Comment by mathildecaron31** [comment]
Thanks @woctezuma for the detailed response :).

I close the issue, feel free to reopen if needed @abeyang00
